coral-schema/src/main/java/com/linkedin/coral/schema/avro/SchemaUtilities.java

```diff
@@ -273,20 +278,63 @@ static String getFieldName(String oldName, String suggestedNewName) {
     return newName;
   }
 
-  static String generateDocumentationForLiteral(RexLiteral rexLiteral) {
+  private static String getLiteralValueAsString(@Nonnull RexLiteral rexLiteral) {
     StringWriter documentationWriter = new StringWriter();
     PrintWriter printWriter = new PrintWriter(documentationWriter);
 
     rexLiteral.printAsJava(printWriter);
     printWriter.flush();
 
-    return "Field created from view literal with value: " + documentationWriter;
+    return documentationWriter.toString();
   }
 
-  static String generateDocumentationForAggregate(AggregateCall aggregateCall) {
+  /**
+   * Given an input {@link RelNode} and the index of a field in the {@link RelNode}'s corresponding Avro schema,
+   * determine if the field with the specified index is a column from a table.
+   * @param fieldIndex the index of a field in the <code>inputRelNode</code>'s corresponding Avro schema
+   * @param inputRelNode the input {@link RelNode}
+   * @return true if the field at <code>fieldIndex</code> is a column from a table
+   */
+  private static boolean isColumn(int fieldIndex, @Nonnull RelNode inputRelNode) {
+    return !(inputRelNode instanceof LogicalAggregate)
+        || fieldIndex < ((LogicalAggregate) inputRelNode).getGroupSet().cardinality();
+  }
```

Comment:
Given an input `RelNode` and a field index, how can we generalize the logic for determining whether or not the field is referring to a column from a table? I don't think this current approach is very robust.

@funcheetah @ljfgem wondering if you have any suggestions? 🙂

Reply:
I would prefer to determine it by:
```
return !inputRelNode.getRowType().getFieldNames().get(fieldIndex).contains("$");
```
`$` wouldn't appear in a normal column name, but would appear in the auto-assigned column name in the row type.

Reply:
Thank you, I'll give this a try!

Reply:
@ljfgem This seems to work for all current test cases, but it looks like from Hive 0.13 onwards, `$` can be a valid character in a column name.

From Apache Hive [docs](https://cwiki.apache.org/confluence/display/hive/languagemanual+ddl#LanguageManualDDL-CreateTable):
>* In Hive 0.12 and earlier, only alphanumeric and underscore characters are allowed in table and column names.
>* In Hive 0.13 and later, column names can contain any Unicode character (see HIVE-6013), however, dot (.) and colon (:) yield errors on querying, so they are disallowed in Hive 1.2.0 (see HIVE-10120). Any column name that is specified within backticks (`) is treated literally. Within a backtick string, use double backticks (``) to represent a backtick character. Backtick quotation also enables the use of reserved keywords for table and column identifiers.

Reply:
Hmmm, then I am good with the current logic for determining it.

Reply:
Sounds good. Are there any special cases other than `LogicalAggregate` that I should account for?

Reply:
Nothing on top of my mind at present, we can add if we encounter a corner case in the future.

Reply:
I'm okay with this approach to determine whether it is a column.